avocado/core/loader.py

```diff
@@ -684,6 +690,77 @@ def _find_avocado_tests(self, path, class_name=None):
                                                                       cl_tags),
                                     'path': path}
                             result[statement.name] = info
+                            continue
+
+                # Looking for the 'recursive' docstring or a class_name (what
+                # means we are under recursion)
+                if (safeloader.check_docstring_directive(docstring, 'recursive')
```

Comment:
The recursive should take precedence to `{mod,test}_import`, because you can have multiple inheritance and still inherit from Test.